### PR TITLE
Model default data is not validated during commit

### PR DESCRIPTION
**Pull request by fauvel**
Data field values are only validated when setting the value. 
It's possible to commit invalid (empty) value when commiting without `_set`.

### CODE REVIEW

**Comment by fauvel** [comment]
@tomaszdurka please review.

**Comment by tomaszdurka** [comment]
Hmm, I could have poorly describe my pull-request. 

Actually refactoring you did doesn't cover problem I've encountered.
I meant that if you create empty model object and do not run any kind of setData on it it won't validate. Even if you commit.

Example code:
```php

$model = new CM_Model_Location_City();
$model->commit();

$row = CM_Db_Db::select($model->getTableName(), '*', ['id' => $model->getId()])->fetch();
var_dump($row);
```
Above could dumps:
```
array(7) {
  'id' =>
  string(6) "367989"
  'stateId' =>
  NULL
  'countryId' =>
  string(1) "0"
  'name' =>
  string(0) ""
  'lat' =>
  NULL
  'lon' =>
  NULL
  '_maxmind' =>
  NULL
}
```
`countryId` is `0` while schema marked it as non-optional model: `'countryId' => array('type' => 'CM_Model_Location_Country'),`

**Comment by fauvel** [comment]
@tomaszdurka thanks for clarifying. Please review!

**Comment by fauvel** [comment]
@tomaszdurka I've covered the use case you had in mind, finally :) Please review!

**Comment by tomaszdurka** [comment]
Hey @fauvel can you take a look at tests?

**Comment by fauvel** [comment]
Timeout on Travis

File: library/CM/Model/Abstract.php [hunk 1]
```diff
@@ -436,12 +434,20 @@ protected function _getSchemaData($data = null) {
...
-                $schema->validateField($key, $value);
+            if ($checkMissingFields) {
+                foreach ($schema->getFieldNames() as $key) {
+                    $value = isset($data[$key]) ? $data[$key] : null;
+                    $schema->validateField($key, $value);
+                }
+            } else {
+                foreach ($data as $key => $value) {
+                    $schema->validateField($key, $value);
+                }
```
Comment (tomaszdurka) [inline library/CM/Model/Abstract.php]: What about checking all the fileds always?
I guess those which are optional should stay valid, all the others should be invalid if empty.

Comment (fauvel) [inline library/CM/Model/Abstract.php]: When creating a new model and setting single fields initially, you obviously don't want to do that...

**Comment by fauvel** [comment]
@tomaszdurka good to merge?

**Comment by tomaszdurka** [comment]
Let me check again quickly.

**Comment by fauvel** [comment]
*bump* :)

**Pull request merged**